### DISCUSSION TITLE
Resize Image Feature

### DISCUSSION THREAD

**Discussion by ghost** [outline/outline#1369] [General] [answered]
I am writing a doc but unable to resize the image. It would be nice if we can resize images in outline. 

![Screenshot 2020-07-19 at 10 25 14 AM](https://user-images.githubusercontent.com/29978858/87867458-7a32e300-c9aa-11ea-9883-f344f38d52d5.png)

**Comment by tommoor** [comment, marked as answer]
Any thoughts on the kind of resizing options that would work well for you?

**Reply by ghost** [reply]
![resize](https://user-images.githubusercontent.com/29978858/90340258-81362b00-e014-11ea-8e22-5fb8fbc414a6.gif)

**Reply by RickCogley** [reply]
Not my question, but this is also something we are looking for. If not fancy drag and drop, even a simple selector for percentage resize would be appreciated.

**Reply by dkkb** [reply]
🤔

**Comment by reidab** [comment]
I'd second the notion that some form of image resizing or size specification would be very useful. It's incredibly frustrating to have every screenshot that I take on a screen with a DPR > 1 get pasted in at a giant size.

An alternative to providing resize controls might be to decode embedded DPI info on paste, generate a scaled size automatically from that, and then render the image with an appropriate size constraint.

**Reply by tommoor** [reply]
We did have a version of the DPI decode early on, it never made it in unfortunately as it didn't seem reliable enough. Seems like an ideal thing for a library (the DPI decoding), maybe someone has taken that on my now.

**Reply by cliftonc** [reply]
Speaking from experience, it is much better to allow the images to be uploaded as-is (perhaps with a size limit), and then use a service like imgix.com to resize and cache the images on the fly, and there are now a range of equivalent services that you can self-host like https://github.com/lipojs/lipo.  I really wouldn't want the cost of image manipulation inside the main app slowing things down.  

This could easily be added as a configuration option to the richtext editor (passed in from the hosting app), and it can easily re-write the image urls when rendering the markdown for images using that config, this also then allows you to easily control the size of the image in the page in the same way as the placement of them as per outline/rich-markdown-editor#345.

**Comment by cliftonc** [comment]
This feature is also important to us, and I want to migrate a lot of things to Outline.  So I created a proof of concept that takes the simplest path there, happy to amend / discuss / stop if I'm heading in the complete wrong direction for the future vision of the product.

https://github.com/outline/rich-markdown-editor/pull/345

**Comment by RRoggia** [comment]
I'm also interested in this functionality. Do you guys already have plans to add it to the cloud version?

If you could share a little more on the status, I'm also willing to help if necessary.

**Comment by dkkb** [comment]
+1

**Comment by dkkb** [comment]
What about using image tag to accomplish this?

https://support.typora.io/Resize-Image/

**Comment by haxplorer** [comment]
+1 Without this, it is near impossible to use outline like people use notion - to share posts to external stakeholders, for informational pages, etc.

**Comment by ReessKennedy** [comment]
+1 ... This is the current, major annoyance we're facing with outline over Notion or even something like LogSeq which is also markdown-based and has a cool, very flexible resize on image uploads. We've tried to use the left align to decrease size but that creates margins/padding around images. I like animation from @arpitchaudhari and this resembled the behavior in logseq which appends sizing dynamically as you resize the image. 
<img width="545" alt="Screen Shot 2022-07-27 at 2 29 38 PM" src="https://user-images.githubusercontent.com/3970267/181346440-e833be83-cde2-4f08-a884-8abd7a5bbcb0.png">

**Reply by ReessKennedy** [reply]
Happy to help with this as well. In my humble opinion, a feature like this goes further to advance adoption and utility than new integrations--just as I see that integrations are getting lots of attention but interest in this was expressed a couple years ago. Still, so many other great things about Outline--would love to see a gap like this be filled.

**Comment by haxplorer** [comment]
Would second @ReessKennedy on this.

**Comment by jazzmonger** [comment]
Such a basic function of every discussion board on the planet.  Except here....

**Reply by tommoor** [reply]
Included in `v0.67.0+`

**Comment by tommoor** [comment]
This feature is in included in `v0.67.0+`
